addons/sale/models/sale.py

```diff
@@ -544,6 +544,7 @@ def _prepare_invoice(self):
             'campaign_id': self.campaign_id.id,
             'medium_id': self.medium_id.id,
             'source_id': self.source_id.id,
+            'user_id': self.user_id and self.user_id.id,
```

Comment:
No need that repetition

```suggestion
            'user_id': self.user_id.id,
```